chart/templates/tests/test-metrics.yaml

```diff
@@ -0,0 +1,150 @@
+{{- $tsdb := index .Values "timescaledb-single" -}}
+{{- $kubePrometheus := index .Values "kube-prometheus-stack" -}}
+{{- if or $tsdb.enabled $kubePrometheus.enabled -}}
+---
+apiVersion: v1
+kind: ServiceAccount
+automountServiceAccountToken: false
+metadata:
+  name: "{{ .Release.Name }}-test-metrics"
+  namespace: "{{ template "tobs.namespace" . }}"
+  labels:
+    {{- include "tobs.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": test
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: "{{ .Release.Name }}-test-metrics"
+  namespace: "{{ template "tobs.namespace" . }}"
+  labels:
+    {{- include "tobs.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": test
+data:
+  test-metrics.sh: |-
+    #!/bin/bash
+
+    set -euo pipefail
+
+    # Setup query to prometheus
+    RELEASE="{{ .Release.Name }}"
+    NAMESPACE="{{ .Release.Namespace }}"
+
+{{ if $tsdb.enabled }}
+    PROM_QUERY_URL="http://${RELEASE}-promscale.${NAMESPACE}.svc:9201/api/v1/query"
+{{ else }}
+    PROM_QUERY_URL="http://${RELEASE}-kube-prometheus-stack-prometheus.${NAMESPACE}.svc:9090/api/v1/query"
+{{ end }}
+
+    testset=$(cat <<-EOF
+    [
+{{ if $kubePrometheus.enabled }}
+      {
+        "expression": "alertmanager_build_info{namespace=\"$NAMESPACE\"}",
+        "expected": true
+      },
+      {
+        "expression": "node_exporter_build_info{namespace=\"$NAMESPACE\"}",
+        "expected": true
+      },
+      {
+        "expression": "prometheus_build_info{namespace=\"$NAMESPACE\"}",
+        "expected": true
+      },
+      {
+        "expression": "prometheus_operator_build_info{namespace=\"$NAMESPACE\"}",
+        "expected": true
+      },
+{{ end }}
+{{ if .Values.promscale.enabled }}
+      {
+        "expression": "promscale_build_info{namespace=\"$NAMESPACE\"}",
+        "expected": true
+      },
+{{ end }}
```

Comment:
Those blocks allow us to easily parametrize tests based on `values`/modules used by end-user.